### PR TITLE
Feature/add override joystick commander

### PR DESCRIPTION
**Pull request by LucasBuckland**
Waiting on validation from @evanlivingston but wanted to get the PR up for a code review.

### CODE REVIEW

File: control/joystick_commander/include/can_read.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/************************************************************************/
+/* The MIT License (MIT) */
+/* ===================== */
+
+/* Copyright (c) 2016 PolySync Technologies, Inc.  All Rights Reserved. */
```
Comment (ltbj) [inline control/joystick_commander/include/can_read.h]: Need to update our copyrights, but we have a tool that can sweep all files - we can get that in another commit.

File: control/joystick_commander/src/can_read.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        }
+    }
+    
+    return ret;
+}
+
+
+//
+int can_read_buffer( 
+        commander_s * const commander )
```
Comment (ltbj) [inline control/joystick_commander/src/can_read.c]: Need to check this pointer is valid before referencing it.

File: control/joystick_commander/src/can_read.c [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        // Do nothing
+    }
+    else
+    {
+        ret = ERROR;
+    }
+
+
+    return ret;
+}
```
Comment (ltbj) [inline control/joystick_commander/src/can_read.c]: Missing the newline at the end of file.

File: control/joystick_commander/src/messages.c [hunk 4]
```diff
@@ -207,4 +207,4 @@ int messages_set_default_values(
 
 
     return ret;
-}
+}
```
Comment (ltbj) [inline control/joystick_commander/src/messages.c]: Missing the newline at the end of file.

File: control/joystick_commander/src/can_read.c [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    if( (steering_report == NULL) || (driver_override_flag == NULL) )
+    {
+        ret = ERROR;
+    }
+    
+    if( ret == NOERR )
+    {
+        if( (*driver_override_flag) == 0 )
+        {
+            (*driver_override_flag) = steering_report->override;
```
Comment (p6363091) [inline control/joystick_commander/src/can_read.c]: do you need to cast this value to type int?

Comment (LucasBuckland) [inline control/joystick_commander/src/can_read.c]: Good point, will fix this.

File: control/joystick_commander/src/messages.c (hunk 4 above)
Comment (LucasBuckland) [inline control/joystick_commander/src/messages.c]: Good catch, will fix.

File: control/joystick_commander/src/can_read.c (hunk 3 above)
Comment (LucasBuckland) [inline control/joystick_commander/src/can_read.c]: Good catch, will fix.

File: control/joystick_commander/include/can_read.h (hunk 1 above)
Comment (LucasBuckland) [inline control/joystick_commander/include/can_read.h]: So leave this?

File: control/joystick_commander/src/can_read.c (hunk 2 above)
Comment (LucasBuckland) [inline control/joystick_commander/src/can_read.c]: This is actually done in the commander_is_valid function which is called prior to can_read_buffer.

**Pull request merged**